livekit-agents/livekit/agents/tts/fallback_adapter.py

```diff
@@ -0,0 +1,507 @@
+import asyncio
+from typing import Literal, Union, AsyncGenerator
+from livekit import rtc
+import dataclasses
+import contextlib
+
+from pydantic.type_adapter import P
+
+from .tts import (
+    TTS,
+    ChunkedStream,
+    SynthesizedAudio,
+    SynthesizeStream,
+    TTSCapabilities,
+    APIConnectOptions,
+    DEFAULT_API_CONNECT_OPTIONS,
+)
+
+from .. import utils
+from ..utils import aio
+from .._exceptions import APIConnectionError, APIError
+
+from ..log import logger
+
+from dataclasses import dataclass
+
+
+# don't retry when using the fallback adapter
+DEFAULT_FALLBACK_API_CONNECT_OPTIONS = APIConnectOptions(
+    max_retry=0, timeout=DEFAULT_API_CONNECT_OPTIONS.timeout
+)
+
+
+@dataclass
+class _TTSStatus:
+    available: bool
+    recovering_task: asyncio.Task | None
+    resampler: rtc.AudioResampler | None
+
+
+@dataclass
+class AvailabilityChangedEvent:
+    tts: TTS
+    available: bool
+
+
+class FallbackAdapter(
+    TTS[Literal["tts_availability_changed"]],
+):
+    def __init__(
+        self,
+        tts: list[TTS],
+        *,
```

Comment:
a syntactic sugar idea i had when i wrote this a few months back:

```suggestion
        *tts: list[TTS],
```

that way you can write `FallbackAdapter(tts1, tts2, tts3, timeout=5.0)`

Reply:
it looks better but this issue with this synthetic sugar, is that the parameter is optional